src/ray/core_worker/core_worker.cc

```diff
@@ -229,13 +229,22 @@ void CoreWorker::RunIOService() {
 void CoreWorker::SetCurrentTaskId(const TaskID &task_id) {
   worker_context_.SetCurrentTaskId(task_id);
   main_thread_task_id_ = task_id;
+  // NOTE(zhijunfu): AsyncUnsubscribe() is async and cannot guarantee
+  // that no actor update notifications will be received after it's called.
+  // If the worker receives a task which again adds actor handles for the
+  // same actors, then it's possible that there are two callbacks invoked
+  // for the same actor ID, which would crash when the second one tries to
+  // append the rpc client for the actor. So disable the logic below until
```

Comment:
Thanks! But why not just keep the original rpc client for now?

Reply:
Could you please explain more on your suggestion. The issue is that `ConnectAndSendPendingTasks` is called twice and at the second time it crashes before there's already a rpc client for this `actor_id` thus the emplace fails.

Reply:
Right, I was saying that we could just keep the original rpc client instead of trying to do the emplace.

Anyway, I think this is fixed now that we merged #6191. You can see the code [here](https://github.com/ray-project/ray/blob/master/src/ray/core_worker/transport/direct_actor_transport.cc#L120). Can you uncomment this and see if it works now?

Reply:
Reverted.